packages/cursorless-engine/src/processTargets/modifiers/scopeHandlers/TokenScopeHandler.ts

```diff
@@ -42,12 +42,28 @@ export class TokenScopeHandler extends NestedScopeHandler {
     } = scopeA;
     const { identifierMatcher } = getMatcher(document.languageId);
 
+    const textA = document.getText(scopeA.domain);
+
     // NB: Don't directly use `test` here because global regexes are stateful
     // See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec#finding_successive_matches
-    return testRegex(identifierMatcher, document.getText(scopeA.domain))
-      ? true
-      : testRegex(identifierMatcher, document.getText(scopeB.domain))
-      ? false
-      : undefined;
+    if (testRegex(identifierMatcher, textA)) {
```

Comment:
I'm also a bit surprised by our logic here. Why do we immediately prefer `a` if it's an identifier; what if `b` is also an identifier? I would expect us to return `undefined` in that case 🤔. I guess you're just mirroring the same logic here but seems wrong to me

Reply:
Ohh right yeah I guess we never hit that with two identifiers

False and undefined are different I believe. False means prefer b; undefined means no preference so let modifier decide

I would argue we should be correct here now that we actually can have them adjacent

Reply:
I'm just saying that today we don't actually utilize that. We could simplify the function signature and still have the same behavior yesterday.

Reply:
But I have now updated the code to reflect the actual logic